tests/test_factory_fields.py

```diff
@@ -121,3 +121,41 @@ class MyFactory(ModelFactory):
         assert result.caption == "this was really long for me"
     else:
         assert result.caption == "just this"
+
+
+def test_post_generation_classmethod() -> None:
+    class MyModel(BaseModel):
+        from_dt: datetime
+        to_dt: datetime
+        is_long: bool
+        caption: str
+
+    class MyFactory(ModelFactory):
+        __model__ = MyModel
+
+        random_delta: timedelta
+
+        @PostGenerated
+        @classmethod
+        def to_dt(cls, from_dt: datetime) -> datetime:
+            # save it to cls for test purposes only
+            cls.random_delta = timedelta(days=cls.__random__.randint(0, 12), seconds=cls.__random__.randint(13, 13000))
+            return from_dt + cls.random_delta
+
+        @PostGenerated
```

Comment:
Nice addition! 

I suggest you simply create a separate decorator for this or alias it to standard snake case as a decorator. 

Also required are docs

Reply:
Thanks! Any naming suggestions? `post_generated` or something else?

Also if we're going with separate decorator, we can do without the extra `@classmethod`. So either
```py
    @post_generated
    @classmethod
    def to_dt(cls, from_dt): ...
```
or
```py
    @post_generated
    def to_dt(cls, from_dt): ...
```
can be made to work. (1) is more explicit, (2) more succinct. I don't have any strong preference, lmk and i'll update the PR (+docs).

Reply:
I think the top one.

Reply:
Code & docs were updated as suggested.